app/src/org/commcare/logging/DataChangeLogger.kt

```diff
@@ -0,0 +1,156 @@
+package org.commcare.logging
+
+import android.content.Context
+import android.os.Environment
+import android.util.Log
+import com.crashlytics.android.Crashlytics
+import org.apache.commons.lang3.StringUtils
+import org.commcare.android.logging.ReportingUtils
+import org.commcare.utils.CrashUtil
+import org.commcare.utils.FileUtil
+import org.javarosa.core.io.StreamsUtil
+import java.io.File
+import java.io.FileInputStream
+import java.io.FileOutputStream
+import java.io.IOException
+import java.text.SimpleDateFormat
+import java.util.*
+
+/**
+ *  Used to log some crtitical platform events like DB migrations, Clear user data,
+ *  app installations etc. All these logs are backed by file storage and are accessible
+ *  from outside CommCare
+ */
+class DataChangeLogger {
+
+    fun init(context: Context) {
+        initLogFiles(context)
+    }
+
+    private fun initLogFiles(context: Context) {
+        if (!isExternalStorageWritable()) {
+            Log.e(TAG, "External Storage unavialable to write logs")
```

Comment:
Let's log this information to the device logs instead. Logging to the console does ~nothing for us in situations of real app usage, and that way we'll be able to refer to the device logs to know if we shouldn't expect a user's device to have Data Change Logs on it.